lighthouse-core/test/gather/driver-test.js

```diff
@@ -305,6 +305,123 @@ describe('.evaluateAsync', () => {
   });
 });
 
+describe('.evaluate', () => {
+  it('transforms parameters into an expression given to Runtime.evaluate', async () => {
+    connectionStub.sendCommand = createMockSendCommandFn()
+      .mockResponse('Runtime.evaluate', {result: {value: 1}});
+
+    /** @param {number} value */
+    function main(value) {
+      return value;
+    }
+    const value = await driver.evaluate(main, {args: [1]});
+    expect(value).toEqual(1);
+
+    const {expression} = connectionStub.sendCommand.findInvocation('Runtime.evaluate');
+    const expected = `
+(function wrapInNativePromise() {
+        const __nativePromise = globalThis.__nativePromise || Promise;
+        const URL = globalThis.__nativeURL || globalThis.URL;
+        return new __nativePromise(function (resolve) {
+          return __nativePromise.resolve()
+            .then(_ => (() => {
+      
+      function main(value) {
+      return value;
+    }
+      return main(1);
+    })())
+            .catch(function wrapRuntimeEvalErrorInBrowser(err) {
+  err = err || new Error();
+  const fallbackMessage = typeof err === 'string' ? err : 'unknown error';
+  return {
+    __failedInBrowser: true,
+    name: err.name || 'Error',
+    message: err.message || fallbackMessage,
+    stack: err.stack || new Error().stack
+  };
+})
+            .then(resolve);
+        });
+      }())`.trim();
+    expect(expression).toBe(expected);
+    expect(await eval(expression)).toBe(1);
+  });
+
+  it('transforms parameters into an expression (basic)', async () => {
+    const mockFn = driver._evaluateInContext = jest.fn()
+      .mockImplementation(() => Promise.resolve());
+
+    /** @param {number} value */
+    function mainFn(value) {
+      return value;
+    }
+    await driver.evaluate(mainFn, {args: [1]});
+
+    const code = mockFn.mock.calls[0][0];
+    expect(code).toBe(`(() => {
+      
+      function mainFn(value) {
+      return value;
+    }
+      return mainFn(1);
+    })()`);
+    expect(eval(code)).toEqual(1);
+  });
+
+  it('transforms parameters into an expression (complex)', async () => {
+    const mockFn = driver._evaluateInContext = jest.fn()
+      .mockImplementation(() => Promise.resolve());
+
+    /**
+     * @param {{a: number, b: number}} _
+     * @param {any} passThru
+     */
+    function mainFn({a, b}, passThru) {
+      return {a: abs(a), b: square(b), passThru};
+    }
+    /**
+     * @param {number} val
+     */
+    function abs(val) {
+      return Math.abs(val);
+    }
+    /**
+     * @param {number} val
+     */
+    function square(val) {
+      return val * val;
+    }
+
+    await driver.evaluate(mainFn, {
```

Comment:
possibly silly overkill idea: since this file is type checked, this could also serve a test for the return type as well (even though the return _value_ will be meaningless in this case). Something like

```suggestion
    /** @type {{a: number, b: number, passThru: any}} */
    const result = await driver.evaluate(mainFn, { // eslint-disable-line no-unused-vars
```